### PR TITLE
Introduce KeplerTable class

### PR DESCRIPTION
**Pull request by ibgreen**
# Starts implementing RFC #1109 

This has remaining one failing test case in a reducer dealing with split maps - it is a little hard to follow the logic around what it is doing there, but the error indicates that it might be trying to apply a spread operator to the class instance.

```
  Failed Tests: There was 1 failure

    #visStateReducer -> UPDATE_VIS_DATA.1 -> No data

      ✖ TypeError: Invalid attempt to spread non-iterable instance

```

### CODE REVIEW

**Comment by macrigiuseppe** [comment]
@ibgreen if you want to debug a test case, simply temporarily remove tap-spec from test-node yarn command. You can also run a single test by adding __.only__ to your test case

**Comment by ibgreen** [comment]
@macrigiuseppe Good suggestions, I did remove tap-spec while I fixed a bunch of other issues. 

I am a bit spoiled, I am used to being able to run the test suites directly in the browser (via ocular-dev-tools) and set breakpoints etc.  

But I suppose I can modify the scripts to fire up the node debugger and connect chrome to that.

**Comment by macrigiuseppe** [comment]
@ibgreen you can debug to directly into your IDE the same way you to it with the browser

**Comment by ibgreen** [comment]
> you can debug to directly into your IDE the same way you to it with the browser

Yeah, but since this doesn't start using node (it starts using babel-tape-runner) I am no longer sure how to make it work...

**Comment by macrigiuseppe** [comment]
it's pretty much the same thing but it helps if you have a nice IDE that does it automatically for you

**Comment by ibgreen** [comment]
> it's pretty much the same thing but it helps if you have a nice IDE that does it automatically for you

Yes I'd love to get that working. 

Anyway, good old trial and error worked, bug fixed.

File: package.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
     "fast-test": "npm run test-node && npm run test-browser",
     "test-browser": "NODE_ENV=test babel-tape-runner -r ./test/setup-browser-env.js ./test/browser.js | tap-spec",
     "test-node": "NODE_ENV=test babel-tape-runner ./test/node.js | tap-spec",
+    "test-debug": "NODE_ENV=test babel-tape-runner ./test/node.js",
```
Comment (macrigiuseppe) [inline package.json:27]: don't forget to undo this change was you are done

**Comment by ibgreen** [comment]
I don't want these to auto deploy so I am merging into a feature branch. @macrigiuseppe will help me add some e2e tests before we merge to master.
